f600233d1b154d507b782f84604bcb12/update/sys_script_include_3a155edd97ed11102b1ef0b0f053aff0.xml

```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update table="sys_script_include">
+    <sys_script_include action="INSERT_OR_UPDATE">
+        <access>package_private</access>
+        <active>true</active>
+        <api_name>x_g_dis_atat.FundingService</api_name>
+        <caller_access/>
+        <client_callable>false</client_callable>
+        <description>Grouped funding functionality used within ATAT (e.g., Alerts).</description>
+        <name>FundingService</name>
+        <script><![CDATA[var FundingService = Class.create();
+FundingService.prototype = {
+    initialize: function() {
+		this.TO_TABLE = "x_g_dis_atat_task_order";
+		this.CLIN_TABLE = "x_g_dis_atat_clin";
+    },
+	getFollowOnCLIN: function(clinRec) {
+		// find the related TO for the CLIN
+		var toRecord = new GlideRecord(this.TO_TABLE);
+		toRecord.addQuery("clins", "CONTAINS", clinRec.sys_id);
+		toRecord.query();
+		toRecord.next();
+
+		// get all CLIN records for the task order
+		var clinRecord = new GlideRecord(this.CLIN_TABLE);
+		clinRecord.addQuery("sys_id", "IN", toRecord.clins);
+		clinRecord.addQuery("clin_status", "!=", "EXPIRED");
+		clinRecord.query();
+		
+		var toClins = [];
+		while (clinRecord.next()) {
+           // attempted to pass along only the clin records which would reduce having to do another query 
+           // later, but there was some weird behavior and only the first clin Record would populate in the
+           // array (e.g., toClins) when looping over the records in the "while" loop above. Not sure if
+           // there is a special way to do it or if that is a limitation with in SN.
+			toClins.push(clinRecord.clin_number.toString());
+		}
+		
+		gs.info("Task Order CLINS found: " + toClins);
+		var followOnClins = toClins
+			.filter(function(clinNumber) { 
+				// exclude current clin
+				return clinNumber !== clinRec.clin_number.toString(); } )
+            .filter(function(clinNumber) { 
+				// get clins that are in future periods
+                // assumes the clin first char identifies the period
+                var currentPeriod = clinRec.clin_number.toString()[0]++;
+                var selectedClinPeriod = clinNumber[0]++; 
+
+               // period ceiling is 9 currently
+               return selectedClinPeriod > currentPeriod && selectedClinPeriod <= 9;  })
```

Comment:
@jasonburkert-ccpo @jeffsegal-ccpo what are we considering a follow-on CLIN? 
- Is it any CLIN that is in a future period? (e.g.,`1XXX` -> `2XXX`) 
OR
- A CLIN that is in a future period and matches? (e.g., `1001` -> `2001`)

Reply:
The latter. That is, 2020 is not a follow-on to 1017.

Reply:
That's what I was thinking, but wanted to make sure. I'll make that change. I'll also rebase and update the `Evaluate Time Remaining for CLIN` to include the script include.

Reply:
Updated `getFollowOnCLIN` to have matching CLINS as mentioned above. Changes in https://github.com/dod-ccpo/atat-snow/pull/127/commits/1784740f87deefe62ba3f32fcdae744bad31d436.